internal/api/courses/assignments/submissions/submit_test.go

```diff
@@ -24,6 +26,7 @@ func TestSubmit(test *testing.T) {
 		fields := map[string]any{
 			"course-id":     testSubmission.Assignment.GetCourse().GetID(),
 			"assignment-id": testSubmission.Assignment.GetID(),
+			"allow-late":    true,
```

Comment:
I'm not a fan of this change,
but I will let it stand as long as it is commented.

However, the test for allow-late should be more targeted and specific
(this this more blanket approach).
I recommend modifying TestRejectLateSubmission() to loop over test cases where the due date is an option for the test case.